### ISSUE TITLE
MySQL Exception when trying to query tickets purchased by customer

### ISSUE THREAD

**Issue by mark-adigital** [verbb/Events#93]
Hi - I'm trying to get the number of tickets purchased by a customer, per [the documentation](https://verbb.io/craft-plugins/events/docs/getting-elements/ticket-queries#customer):

`{% set intTicketsBought = craft.events.tickets().customer(craft.commerce.getCarts().getCart().customer).count() %}`

However, this throws a database exception:

> SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'site-name.elements_sites.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
> The SQL being executed was: SELECT COUNT(*)
> FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`
> FROM `elements` `elements`
> INNER JOIN `events_tickets` `events_tickets` ON `events_tickets`.`id` = `elements`.`id`
> INNER JOIN `commerce_lineitems` `lineitems` ON `events_tickets`.`id` = `lineitems`.`purchasableId`
> INNER JOIN `commerce_orders` `orders` ON `lineitems`.`orderId` = `orders`.`id`
> INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
> INNER JOIN `content` `content` ON `content`.`elementId` = `elements`.`id`
> WHERE (`orders`.`customerId` = '1') AND (`orders`.`isCompleted` = TRUE) AND (`elements`.`archived`=FALSE) AND ((`elements`.`enabled`=TRUE) AND (`elements_sites`.`enabled`=TRUE)) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)
> GROUP BY `events_tickets`.`id`) `subquery`
> INNER JOIN `events_tickets` `events_tickets` ON `events_tickets`.`id` = `subquery`.`elementsId`
> INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
> INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
> INNER JOIN `content` `content` ON `content`.`id` = `subquery`.`contentId`

**Additional info**
- Plugin version: 1.4.19.2
- Craft version: 3.7.8
- Multi-site: No

I see a similar error using `.all()` instead of `.count()` so don't _think_ this is something I'm doing. Thanks!

**Comment by engram-design** [comment]
Can you let me know your MySQL version? This would be an issue with the `sql_mode=only_full_group_by`.

I do actually have `ONLY_FULL_GROUP_BY` enabled for my install just to check, and it's still not throwing this error, so maybe it's coming down to version number differences.

**Comment by mark-adigital** [comment]
Sure - my database client tells me it's `5.7.33-0ubuntu0.16.04.1-log`

**Comment by mark-adigital** [comment]
Hey - just wondered if there was any update on this?

**Comment by engram-design** [comment]
Sorry, I haven't been able to replicate this, also on 5.7.34. I'm pretty positive I have `ONLY_FULL_GROUP_BY` enabled, which is the strict setting that throws this error, so not sure what's going on..

**Comment by engram-design** [comment]
Okay, so If I copy the SQL you've supplied on my end, that does indeed flag it, but it's a different SQL query than mine.

Using the Twig:

```
{{ craft.events.tickets().customer(craft.commerce.getCarts().getCart().customer).count() }}
```

Produces the following SQL.

```
SELECT COUNT(*)
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`
FROM `elements` `elements`
INNER JOIN `events_tickets` `events_tickets` ON `events_tickets`.`id` = `elements`.`id`
INNER JOIN `commerce_lineitems` `lineitems` ON `events_tickets`.`id` = `lineitems`.`purchasableId`
INNER JOIN `commerce_orders` `orders` ON `lineitems`.`orderId` = `orders`.`id`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
INNER JOIN `content` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`)
WHERE (`orders`.`customerId` = '1') AND (`orders`.`isCompleted` = TRUE) AND (`elements`.`type`='verbb\\events\\elements\\Ticket') AND (`elements_sites`.`siteId`=1) AND (`elements`.`archived`=FALSE) AND ((`elements`.`enabled`=TRUE) AND (`elements_sites`.`enabled`=TRUE)) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)
GROUP BY `events_tickets`.`id`) `subquery`
INNER JOIN `events_tickets` `events_tickets` ON `events_tickets`.`id` = `subquery`.`elementsId`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `content` `content` ON `content`.`id` = `subquery`.`contentId`
```

So I'm thinking maybe its another Twig call on your page?

**Comment by engram-design** [comment]
Aha - its triggering this on a non-multisite, which is bizarre (I would've thought the reverse, and it not make a difference)!

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/events` requirement in `composer.json` to:

```
"require": {
  "verbb/events": "dev-craft-3 as 1.4.19.2",
  "...": "..."
}
```

Then run `composer update`.

**Comment by mark-adigital** [comment]
Great, thanks!

**Comment by engram-design** [comment]
Fixed in [1.4.20](https://github.com/verbb/events/releases/tag/1.4.20)
